mods/persistence/modules/overmap/stellar_anchor.dm

```diff
@@ -1,42 +1,125 @@
-#define BASE_TURF_UPKEEP_COST 50
+#define BASE_TURF_UPKEEP_COST 	  50
+#define MAX_SHIP_TILES		  	 300
+#define MAX_ANCHORED_NAME_LENGTH  50
+GLOBAL_LIST_EMPTY(stellar_anchors)
 
 /obj/machinery/network/stellar_anchor
 	name = "stellar anchor"
 	main_template = "stellar_anchor.tmpl"
 
-	var/list/anchored_areas = list()
+	var/list/anchored_areas
+	var/list/errors
+	var/gen_fluff = "sector"
+
+	var/should_launch = FALSE						 // Used to determine when launching blurb should be added to the UI.
+	var/sector_name									 // Name and identifying tag of the created sector, shuttle, ship etc.
+	var/obj/effect/overmap/visitable/created_sector  // Keep track of the created effect to prevent spam creation of sectors.
+	var/sector_type = /obj/effect/overmap/visitable/sector/created
+	var/sector_color = COLOR_WHITE					 // Color of the sector or other objects created by the stellar anchor.
 
 /obj/machinery/network/stellar_anchor/Initialize()
 	. = ..()
 	GLOB.world_saving_event.register(SSpersistence, src, /obj/machinery/network/stellar_anchor/proc/on_world_saving)
+	GLOB.stellar_anchors += src
+	set_extension(src, /datum/extension/eye/stellar_anchor)
 
 /obj/machinery/network/stellar_anchor/Destroy()
 	. = ..()
 	GLOB.world_saving_event.unregister(SSpersistence, src)
+	GLOB.stellar_anchors -= src
 
 /obj/machinery/network/stellar_anchor/proc/on_world_saving()
 	if(is_paid())
 		refresh_anchored_areas()
 		for(var/A in anchored_areas)
 			SSpersistence.AddSavedArea(A)
 
-// returns whether or not the place the stellar_anchor is, is a viable place to actually anchor/persist something.
+/obj/machinery/network/stellar_anchor/ui_data(var/mob/user, ui_key)
+	. = ..()
+	.["anchored_areas"] = anchored_areas
+	.["sector_name"] = sector_name
+	.["should_launch"] = should_launch
+	.["created_sector"] = !!created_sector
+	.["sector_color"] = sector_color
+	if(LAZYLEN(errors))
+		.["errors"] = errors
+
+/obj/machinery/network/stellar_anchor/OnTopic(var/mob/user, href_list, var/datum/topic_state/state)
```

Comment:
Add network settings topic here. You want to get the network_device, and call ui_interact(user) on it.